### ISSUE TITLE
DateRangeFilter crashes if the current locale isn't supported

### ISSUE THREAD

**Issue by diomidov** [ghiscoding/Angular-Slickgrid#346]
### Your Environment

| Software          | Version(s) |
| ----------------- | ---------- |
| Angular           | 7.2.15        | 
| Angular-Slickgrid | 2.12.3        | 
| TypeScript        | 3.1.6        | 
| flatpickr     | 4.6.3        | 
| Operating System  | Windows 10        | 

### Context
My current locale is `ua`. I want to create a column with a date range filter. Flatpickr doesn't support `ua` locale out of the box, so I tried explicitly specifying a supported locale:
```typescript
filter: {
  model: Filters.dateRange,
  filterOptions: {
    locale: 'en',
  } as FlatpickrOption,
},
```

### Expected Behavior
A date range filter that uses English locale

### Current Behavior
```
Error: "Cannot find module './ua.js'"
    webpackContextResolve .*\.js$:73
    webpackContext .*\.js$:67
    Angular 6
        loadFlatpickrLocale
        buildDatePickerInput
        createDomElement
        init
        addFilterTemplateToHeaderRow
        bindLocalOnFilter
    notify slick.core.js:183
    trigger slick.grid.js:2555
    createColumnHeaders slick.grid.js:1303
    finishInitialization slick.grid.js:557
    Angular 37
        initialization
        ngAfterViewInit
        ...
```
(sorry, this doesn't show typescript line numbers)

### Possible Solution
Here is the relevant part of DateRangeFilter.buildDatePickerInput:
```typescript
  //...
  let currentLocale = this.translate && this.translate.currentLang || 'en'; // returns 'ua'
  //...
  const pickerOptions = {
    //...
    // tries to read ua.js and throws an exception because ua.js doesn't exist
    locale: (currentLocale !== 'en') ? this.loadFlatpickrLocale(currentLocale) : 'en',
    //...
  }
  //...
  // override the current locale with the locale from filterOptions
  this._flatpickrOptions = { ...pickerOptions, ...(this.columnFilter.filterOptions as FlatpickrOption) };
  //...
```

I think `loadFlatpickrLocale` should return `en` if it can't load the locale, instead of throwing an exception. Alternatively you can avoid calling `this.loadFlatpickrLocale(currentLocale)` if `pickerOptions.locale` exists.

**Comment by ghiscoding** [comment]
That is a valid bug, I created a PR #347  to address wath you wrote. However please note that I will show a console warning (see the print screen in the PR) to advise user that it's not a supported locale in Flatpickr, I could have use English straight away without any console warning, but I think it's better to advise the user (you) that it's not a valid Flatpickr locale and so you know where the error originated. If you read the warning, I mentioned that you can avoid seeing the console warning by overriding the locale via the `filterOptions`, which is what you wanted. 

See the PR #347 for more info and if you could confirm that this is the behavior you wanted, I'll merge it right after. 

Thanks for reporting 
Cheers ⭐️

**Comment by ghiscoding** [comment]
I'll merge the PR and go with that since I didn't feedback from your side and I need to fix another bug and do another release soon.

**Comment by ghiscoding** [comment]
Released under [version 2.14.3](https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v2.14.3)
